finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala

```diff
@@ -31,6 +31,11 @@ private[finagle] class HttpServerDispatcher(
     service.close()
   }
 
+  private def mayHaveContent(status: Status): Boolean = status match {
```

Comment:
Hi @taylorleese  cc: @mosesn ,
I might be wrong but it seems like [handle(res: Reponse)](https://github.com/monkey-mas/finagle/blob/4002b80fe5b124459beaa8a3c8f364c3cb0762a2/finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala#L72) is called multiple times and it's called again after all [filters](https://github.com/monkey-mas/finagle/blob/4002b80fe5b124459beaa8a3c8f364c3cb0762a2/finagle-http/src/main/scala/com/twitter/finagle/Http.scala#L304-L308) are applied (s.t., even after `ComplianceFilter` is applied) as I mentioned before. I found this when I debug my commit with sample code like [this](https://gist.github.com/monkey-mas/eeade1dc021ab9f96a1dfa5f6198e09d).

I think this means a Content-Length is added to a 204 response for example. Suppose we're going to return a 204 response;
1)we _remove_ a Content-Length and message-body of a 204 response by using `ComplianceFilter`
2)but then `handle` is called and this [line](https://github.com/twitter/finagle/blob/develop/finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala#L96-L97) adds a Content-Length with a value of 0 to a 204 response as we've _already_ removed a Content-Length in 1) and `!rep.contentLength.isDefined` holds.

To achieve what we want to do, i.e., send a 204 without a Content-Length and message-body for example, we need to have this condition [here](https://github.com/monkey-mas/finagle/blob/4002b80fe5b124459beaa8a3c8f364c3cb0762a2/finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala#L101). The condition checks whether or not send a Content-Length.

This is why I think we still need `myHaveContent`. Anyway, I need your help to sort this out. Can you please help me? :)